### ISSUE TITLE
Enhancement: Markdown in Artifacts

### ISSUE THREAD

**Issue by PaulWoitaschek** [danny-avila/LibreChat#3995]
### What features would you like to see added?

I'd like markdown previews to be rendered in artifacts

### More details

Currently they get rendered but not as markdown

### Which components are impacted by your request?

UI

### Pictures

<img width="1067" alt="image" src="https://github.com/user-attachments/assets/44dff2a9-2c85-40b5-a541-2b85a3a3405e">
<img width="1068" alt="image" src="https://github.com/user-attachments/assets/99a55427-78e3-4baa-9f67-efe37efb4b2c">

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by dareima** [comment]
Yep, wanted to report this as well. Stumbled over this when I learned that Artifacts at Claude are also rendering markdown.

A download button to download the .md document would be nice, too (check Claude for reference).

LibreChat rocks!

**Comment by Vinnie-Singleton-NN** [comment]
Just wanted to drop a comment in agreement—this would be a nice feature to have. 

Claude's ability to handle this leads users to try it out and artifacts picking up on the format but not displaying markdown correctly can make it feel like a bug. Appreciate all the work that's being done.

**Comment by wireflo** [comment]
I also came across this just now any made myself totally mad- did a clean rebuild of my setup even because I was totally convinced I saw this working 😅 - obviously confused it with claude.ai
In general small adjustment in the prompting could fix this and additionally apply something like <pre styling to any artifact text output.

**Comment by barronlroth** [comment]
Bumping this! 

I end up having to drop the Markdown in Cursor to preview.
